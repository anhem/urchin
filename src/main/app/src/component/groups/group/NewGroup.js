import React, {Component} from 'react'
import FieldError from "../../FieldError"
import history from '../../../history'

class NewGroup extends Component {

    update = (e) => {
        let data = {
            field: e.target.name,
            value: e.target.value
        };

        this.props.callbacks.setGroup(data);
    };

    create = () => {
        this.props.callbacks.createGroup(this.props.group);
    };

    cancel = () => {
        history.push('/groups');
    };

    render() {
        let groupName = this.props.group.groupName || '';
        return (

            <div id="new-group">
                <h2>Group</h2>
                <input
                    name="groupName"
                    type="text"
                    value={groupName}
                    onChange={this.update}
                />
                <FieldError
                    fieldErrors={this.props.fieldErrors}
                    field="groupName"
                />
                <button id="new-group__create-btn"onClick={this.create}>Create Group</button>
                <button id="new-group__cancel-btn" onClick={this.cancel}>Cancel</button>
            </div>
        )
    }
}

export default NewGroup