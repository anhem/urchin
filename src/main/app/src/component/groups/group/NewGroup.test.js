import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import NewGroup from "./NewGroup";

describe('NewGroup', () => {

    let callbacks = {
        setGroup: jest.fn(),
        createGroup: jest.fn(),
    };

    let props = {
        group: {},
        callbacks: callbacks
    };

    beforeEach(() => {
        Object.entries(callbacks).forEach(([name, fn]) => {
            fn.mockClear();
        });
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<NewGroup {...props}/>))).toMatchSnapshot();
    });

    it('setGroup is called when groupName changes', () => {
        let component = shallow(<NewGroup {...props}/>);
        component.find('[name="groupName"]').simulate('change', {target: {value: "name"}});

        expect(callbacks.setGroup).toHaveBeenCalledTimes(1);
    });

    it('createGroup is called when clicking create', () => {
        let component = shallow(<NewGroup {...props}/>);
        component.find('[data-view="create"]').simulate('click');

        expect(callbacks.createGroup).toHaveBeenCalledWith(props.group)
    });
});
