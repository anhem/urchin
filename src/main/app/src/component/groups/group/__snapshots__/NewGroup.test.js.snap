// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGroup match snapshot 1`] = `
<div
  className="new-group"
  data-view="newGroup"
>
  <h2>
    Group
  </h2>
  <input
    name="groupName"
    onChange={[Function]}
    type="text"
    value=""
  />
  <FieldError
    field="groupName"
  />
  <button
    className="new-group__create-btn"
    data-view="create"
    onClick={[Function]}
  >
    Create Group
  </button>
  <button
    className="new-group__cancel-btn"
    data-view="cancel"
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;
